src/NuGet.Core/NuGet.ProjectModel/PackageSpecOperations.cs

```diff
@@ -26,7 +26,7 @@ public static void AddOrUpdateDependency(PackageSpec spec, PackageDependency dep
 
             if (!existing.Any())
             {
-                AddDependency(spec.Dependencies, dependency.Id, range);
+                AddDependency(spec, spec.Dependencies, dependency.Id, range);
```

Comment:
I think you need to update the above as well.

```cs
foreach (var existingDependency in existing)
            {
                existingDependency.LibraryRange.VersionRange = range;
            }
```

I think you need update the CPM instance as well there, otherwise you're likely to break no-op.

Reply:
That code path is only called for non-PackageReference projects.

https://github.com/NuGet/NuGet.Client/blob/d1b4bb43c13d56a6fa774e0fa61c070902dc27aa/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L2988-L2996

CPM is only supported by PackageReference style projects so we don't need to add the central package version right?

Reply:
I'm not sure what you mean, can you elaborate on what you want me to do?

Reply:
Basically wondering if we should have a validation that says `if ProjectStyle == PackageReference` in a method where PR can't be handled we throw InvalidOperationException.

Reply:
We talked offline and the check in `PackageSpecOperations` checks if CPM is enabled which is only true for `PackageReference` style projects